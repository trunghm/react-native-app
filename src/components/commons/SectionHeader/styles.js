import EStyleSheet from "react-native-extended-stylesheet";
import GlobalStore from "../../../themes/GlobalStore";

const styles = () => EStyleSheet.create({
  container: {
    width: "100%",
    height: 45,
    justifyContent:"center",
    paddingLeft: 10,
    paddingRight:10,
  },

  text:{
    color : GlobalStore.color.primary
  },
});

export default styles;
