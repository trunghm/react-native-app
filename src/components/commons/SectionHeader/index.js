import React, {Component} from 'react';
import styles from "./styles";
import {
  Text,
  View,
} from 'react-native';

export default function SectionHeader({url, text}) {
  return (
    <View style={styles().container}>
      <Text style={styles().text}>{text}</Text>
    </View>);
}


