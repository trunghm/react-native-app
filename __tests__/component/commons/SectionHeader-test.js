/**
 * @format
 */

import 'react-native';
import React from 'react';
import SectionHeader from '../../../src/components/commons/SectionHeader';
import { shallow } from 'enzyme';

// Note: test renderer must be required after react-native.
import EStyleSheet from 'react-native-extended-stylesheet';
import renderer from "react-test-renderer";


const originalLog = console.log;

beforeAll(() => {
  console.log = jest.fn();
  EStyleSheet.build({

  });
});

afterAll(() => {
  console.log = originalLog;
});
describe('Testing  SectionHeader component', () => {

  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <SectionHeader text={"Ex: SectionHeader"}/>
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('renders as expected', () => {
    const wrapper = shallow(
      <SectionHeader text ="Ex: SectionHeader" />
    );

    expect(wrapper).toMatchSnapshot();

  });

});
