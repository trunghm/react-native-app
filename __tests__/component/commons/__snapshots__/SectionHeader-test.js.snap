// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  SectionHeader component renders as expected 1`] = `ShallowWrapper {}`;

exports[`Testing  SectionHeader component renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "height": 45,
      "justifyContent": "center",
      "paddingLeft": 10,
      "paddingRight": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1496bb",
      }
    }
  >
    Ex: SectionHeader
  </Text>
</View>
`;
